bin/postman-test/collections/core-metadata.postman_collection.json

```diff
@@ -9388,7 +9656,522 @@
+						"header": [
+							{
+								"key": "Content-Type",
+								"value": "application/json"
+							}
+						],
+						"body": {
+							"mode": "raw",
+							"raw": "{\r\n    \"id\": \"{{addressableId}}\",\r\n    \"created\": 1475556860463,\r\n    \"modified\": 1475556860463,\r\n    \"origin\": 1471806386924,\r\n    \"name\": \"gasmeter\",\r\n    \"protocol\": \"HTTP\",\r\n    \"address\": \"172.17.0.6\",\r\n    \"port\": 48093,\r\n    \"path\": \"/gasmeter\",\r\n    \"publisher\": \"DELLPUB\",\r\n    \"user\": \"test\",\r\n    \"password\": \"test123\",\r\n    \"topic\": \"kWhData\"\r\n}"
+						},
+						"url": {
+							"raw": "{{baseUrl}}/api/v1/provisionwatcher",
+							"host": [
+								"{{baseUrl}}"
+							],
+							"path": [
+								"api",
+								"v1",
+								"provisionwatcher"
+							]
+						},
+						"description": "Attempt to add a provision watcher using an invalid JSON request body (in this case, an addressable)."
+					},
+					"response": []
+				},
+				{
+					"name": "319 http://localhost:48081/api/v1/provisionwatcher",
+					"event": [
+						{
+							"listen": "test",
+							"script": {
+								"id": "b7204464-81ad-11ea-bc55-0242ac130003",
+								"exec": [
+									"/*",
+									" * Test Case:  /provisionwatcher - 400 Error - POST ",
+									" * Version: Alpha",
+									" * @Author: Brandon Forster",
+									" *",
+									" */",
+									"//Verify the http status code for 400.",
+									"tests[\"Status code is 400\"] = responseCode.code === 400;",
+									"//Verify if response time is less than 200ms.",
+									"tests[\"Response time is less than \"+data.responseTime+\"ms\"] = responseTime < data.responseTime;"
+								],
+								"type": "text/javascript"
+							}
+						}
+					],
+					"request": {
+						"method": "POST",
+						"header": [
+							{
+								"key": "Content-Type",
+								"value": "application/json"
+							}
+						],
+						"body": {
+							"mode": "raw",
+							"raw": "{  \n   \"origin\":1471806386919,\n   \"identifiers\":{  \n      \"MAC\":\"00-05-1B-A1-99-99\",\n      \"HTTP\":\"10.0.0.1\"\n   },\n   \"profile\":{  \n      \"name\":\"variable speed motor profile\"\n   },\n   \"service\":{  \n      \"name\":\"home variable speed motor device service\",\n      \"adminState\": \"unlocked\",\n      \"operatingState\": \"enabled\",\n      \"addressable\": {\n       \"name\": \"ceiling fan variable speed motor address\"\n      }\n   },\n    \"adminState\": \"unlocked\"\n}"
+						},
+						"url": {
+							"raw": "{{baseUrl}}/api/v1/provisionwatcher",
+							"host": [
+								"{{baseUrl}}"
+							],
+							"path": [
+								"api",
+								"v1",
+								"provisionwatcher"
+							]
+						},
+						"description": "Attempt to add a provision watcher using an invalid JSON request body, not specifying a name"
+					},
+					"response": []
+				},
+				{
+					"name": "320 http://localhost:48081/api/v1/provisionwatcher/:id",
+					"event": [
```

Comment:
This testcase returned `500 Internal Server Error` instead of `404` when I tested it with mongo no security docker compose.

Reply:
That's an issue with the REST logic in EdgeX that this is exposing, not the test.

Reply:
https://github.com/edgexfoundry/edgex-go/pull/2512 should address this problem on EdgeX-Go. I believe we can merge this PR as-is.